lib/git-subrepo

```diff
@@ -1349,10 +1349,32 @@ read-gitrepo-file() {
 
   SAY=false OUT=true RUN git config --file="$gitrepo" subrepo.commit
   subrepo_commit=$output
-
-  FAIL=false \
-  SAY=false OUT=true RUN git config --file="$gitrepo" subrepo.parent
-  subrepo_parent=$output
+  
+  # In case of pull using rebase, the field subrepo.parent SHA-1 could be wrong.
+  #FAIL=false \
+  #SAY=false OUT=true RUN git config --file="$gitrepo" subrepo.parent
+  #subrepo_parent=$output
+
+  # Finding the right parent SHA-1
+  SAY=false OUT=true RUN git log --oneline --pretty=format:"%h"  -- "$gitrepo" 
+  gitrepo_commit_list=$output
+  found_comm="Empty"
```

Comment:
This code base likes everything spelled out, I would prefer found_comm be renamed to common_sha or something fully spelled out.